examples/datasets/plot_limo_data.py

```diff
@@ -111,68 +169,250 @@
+# **Note:** We'll only do this for visualization purposes,
+
+name = "phase-coherence"
+factor = 'factor-' + name
+
+# create phase-coherence percentiles
+df = limo_epochs.metadata
+df[factor] = pd.cut(df[name], 11, labels=False) / 10
+# overwrite metadata
+limo_epochs.metadata = df
+
+# color scheme for percentile plot
+colors = {str(val): val for val in np.sort(df[factor].unique())}
+# compute evoked for each phase-coherence percentile
+evokeds = {str(val): limo_epochs[limo_epochs.metadata[factor] == val].average()
+           for val in colors.values()}
+
+# pick channel to plot
+electrodes = ['C22', 'B11']
+# create figures
+for electrode in electrodes:
+    fig, ax = plt.subplots(figsize=(8, 4))
+    plot_compare_evokeds(evokeds,
+                         axes=ax,
+                         ylim=dict(eeg=[-20, 20]),
+                         colors=colors,
+                         split_legend=True,
+                         picks=electrode,
+                         cmap=(name + " Percentile", "magma"))
+    fig.show()
+
+###############################################################################
+# As shown above, there are some considerable differences between the
+# activation patterns evoked by stimuli with low vs. high phase-coherence.
+#
+# In particular, differences appear to be most pronounced along fronto-central
+# and occipital electrodes.
+
+###############################################################################
+# Prepare data for linear regression analysis
+# --------------------------------------------
+#
+# Next, we can test the significance of these differences using linear
+# regression.But before we go on, we'll start by interpolating any missing
+# channels in the LIMO epochs structure. Some subjects in the datasets contain
+# missing channels (stored in ``limo_epochs.info[‘bads’]``) as these were
+# dropped during preprocessing of the data.
+
+limo_epochs.interpolate_bads(reset_bads=True)
+
+# Furthermore, we'll drop the EOG channels (marked by the "EXG" prefix)
+# present in the LIMO epochs structure.
+limo_epochs.drop_channels(['EXG1', 'EXG2', 'EXG3', 'EXG4'])
+
+###############################################################################
+# Define predictor variables and design matrix
+# --------------------------------------------
+#
+# First, we need to create a design matrix containing information about the
+# variables (i.e., predictors) we want to use for prediction of brain
+# activity patterns. For this purpose, we'll use the information contained in
+# the ``limo_epochs.metadata``. Here, we'll explore the effect of
+# phase-coherence as well as the effect of Face A vs. Face B.
+
+# name of predictors + intercept
+predictor_vars = ['face a - face b', 'phase-coherence', 'intercept']
+
+# create design matrix
+design = limo_epochs.metadata[['phase-coherence', 'face']]
 design['face a - face b'] = np.where(design['face'] == 'A', 1, -1)
```

Comment:
Just

```suggestion
design['face a - face b'] = design['face'] == 'A'
```

should work?

Or do it with a `.map` if you want to code it as `-1, 1`.

Reply:
yeah ... I'd like to keep the -1 vs 1 in order to set the the intercept to 0 in the regression, I'll keep the  `np.where()` method for now.